.circleci/config.yml

```diff
@@ -0,0 +1,76 @@
+version: 2
+jobs:
+  setup:
+    docker:
+      - image: circleci/node:8-browsers
+    steps:
+      - checkout
+      - run:
+          name: Determine which projects have changed and trigger the builds
+          command: |
+            ## Indentify head of master
+            git ls-remote origin | awk "/master/ {print \$1}" > master
+            MASTER_COMMIT=`cat master`
+
+            #Determine range of SHAs for diffing files
+            COMMITS="${CIRCLE_SHA1}..${MASTER_COMMIT}"
+
+            #Check the diff on the commits
+            git diff --name-only $COMMITS | cut -d/ -f1 | sort -u > projects
+            echo -e "Modified directories:\n`cat projects`\n"
+
+            # If modified directories contain yarn, or buildkite directories
+            buildall=0
+            for project in `cat projects`; do
+              if [[ ${project} =~ "yarn" || ${project} =~ ".buildkite" ]]; then
+                buildall=1
+                echo -e "Warning. Dependencies or internal configs have change detected."
+                exit 1
+              fi
+            done
+      - restore_cache:
+          key: deps-{{ checksum "yarn.lock" }}
+      - run:
+          name: Install
+          command: yarn install --frozen-lockfile
+      - save_cache:
+          key: deps-{{ checksum "yarn.lock" }}
+          paths:
+            - node_modules
+      - persist_to_workspace:
+          root: .
+          paths: [.]
+  lint:
+    docker:
+        - image: circleci/node:8-browsers
+    steps:
+      - attach_workspace: { at: . }
+      - restore_cache:
+          key: deps-{{ checksum "yarn.lock" }}
+      - run:
+          name: Run Linter (not enforcing for now)
+          command: yarn lint
```

Comment:
Would be possible to run checkout, the check step, install, linter and tests in the same job? Attaching and detaching workspaces it's extremely slow with CI and I don't think we get much running all this in parallel.

Reply:
@CarlosMecha Just updated so it all runs under one job `build_and_test`.